src/xmipp/libraries/py_xmipp/deepDenoising/augmentators.py

```diff
@@ -0,0 +1,138 @@
+      batchY[i] = scipy.ndimage.interpolation.rotate(batchY[i], angle,reshape=False, mode="reflect")
+  return batchX, batchY
+
+def _random_blur( batchX, batchY, sigma_max):
+  for i in range(batchX.shape[0]):
+    if bool(random.getrandbits(1)):
+      # Random sigma
+      sigma = random.uniform(0., sigma_max)
+      batchX[i] = scipy.ndimage.filters.gaussian_filter(batchX[i], sigma)
+      batchY[i] = scipy.ndimage.filters.gaussian_filter(batchY[i], sigma)
+  return batchX, batchY
+
+
+def _mismatch_projection( batchX, batchY, p=0.05):
+  maxMismatch= 1+int(batchX.shape[1]*.01)
+  for i in range(batchX.shape[0]):
+    if np.random.rand()<p:
+      angle = random.uniform(-maxMismatch,-maxMismatch)
+      batchX[i] = scipy.ndimage.interpolation.rotate(batchX[i], angle,reshape=False, mode="reflect")
+  return batchX, batchY
+  
+
+def initPointsRadius(img_shape, aboveInsteadBelow=True, radiusFraction=0.9):
+
+  xv, yv,= np.meshgrid(np.arange( img_shape[0] ), np.arange( img_shape[1] ), indexing="xy")
+  allCoords= np.stack([xv, yv], axis=-1).reshape((-1,2))
+  central_coords= np.array([img_shape[0]//2, img_shape[1]//2] )
+  distMat= np.squeeze( np.sqrt( np.sum((central_coords- allCoords)**2, axis=1))  )
+  distThr= int(radiusFraction*(img_shape[0]/2))
+  if aboveInsteadBelow:
+    return allCoords[ distMat>distThr ].astype( np.int64)
+  else:
+    return allCoords[ distMat<distThr ].astype( np.int64)
+
+  
+def generateReverseNormalizationFunction(img_shape, radiusFraction=0.8):
+  pointsBelow= initPointsRadius(img_shape, aboveInsteadBelow=False, radiusFraction=radiusFraction)
+  def reverseNormalization(imgs, refs):
+    imgsNorm= np.zeros_like(imgs)
+    for i in range(imgs.shape[0]):
+      img, ref= imgs[i], refs[i]
+      innerParRef= ref[ pointsBelow[...,0], pointsBelow[...,1] ]
+      innerParImg= img[ pointsBelow[...,0], pointsBelow[...,1] ]
+      innerMeanRef, innerStdRef= np.mean(innerParRef), np.std(innerParRef)
+      innerMeanImg, innerStdImg= np.mean(innerParImg), np.std(innerParImg)
+      imgsNorm[i]= (img- (innerMeanImg-innerMeanRef))/(innerStdImg/innerStdRef)
+    return imgsNorm
+  return reverseNormalization
+    
+def generateEmptyParticlesFunction(img_shape, prob=0.2):
+
+  cornerSize= 8
+  imgSide= min(img_shape[:2])
+  nTilesPerSide= imgSide//cornerSize
+  
+  pointsAboveRadius= initPointsRadius(img_shape, radiusFraction=0.9)
+
+  def noisifyImg( img):
+    dataSlice= img[ pointsAboveRadius[...,0], pointsAboveRadius[...,1] ] 
+    meanVal= np.mean( dataSlice )
+    stdVal= np.std( dataSlice )
+    noise_img= np.random.normal(meanVal, stdVal, img.shape )
+    return noise_img
+    
+
+  def swapParticlesForEmptyParticles1( batchX, batchY):
+    for i in range(batchX.shape[0]):
+      if np.random.rand()<prob:
+        batchX[i]= noisifyImg( batchX[i] )
+        batchY[i]= -1*np.ones_like( batchY[i] )
+    return batchX, batchY
+  
+  
+  if imgSide%cornerSize!=0:
```

Comment:
indentation issue?

Reply:
This lin corresponds to 
def generateEmptyParticlesFunction

so it is well indented

Reply:
I believe